batch/parsers/testSuiteParserUtils.js

```diff
@@ -0,0 +1,71 @@
+var _ = require('lodash');
+
+export function normalizeTime(rawTime) {
+	let returnValue = 0;
+	if (rawTime) {
+		returnValue = Number.parseFloat(rawTime);
+	}
+	return returnValue.toFixed(2);
+}
+
+function parsedIntOrZero(stringToParse) {
+	return Number.parseInt(stringToParse) || 0;
+}
+export function testSuiteSummary(raw) {
+	return {
+		'tests': parsedIntOrZero(raw.tests),
+		'failures': parsedIntOrZero(raw.failures),
+		'skipped': parsedIntOrZero(raw.skipped),
+		'errors': parsedIntOrZero(raw.errors)
+	};
+}
+
+export function getFailureDataForSingleTest(raw) {
+	return {
+		'type': raw.$.type,
+		'message': raw.$.message,
+		'raw': raw._
+	};
+}
+
+export function tests(raw) {
+	return _(raw).map(function (test) {
+		var current = (test.failure || [])[0] || {$: {}, _: ''};
+
+		return _({
+			'name': test.$.name,
+			'time': normalizeTime(test.$.time),
+			'classname': test.$.classname,
+			'failure': getFailureDataForSingleTest(current)
+		}).tap(function (result) {
+			!test.failure && delete result.failure;
+		}).value();
```

Comment:
This seems unnecessarily complex. If I get it properly, you build an object, then wrap it with `_`, then call `tap` to delete the `failure` property you just assigned.

Why not simply something like this:

```js
let result = {
    name: test.$.name,
    time: normalizeTime(test.$.time),
    classname: test.$.classname,
}

if (test.failure)
    result.failure = getFailureDataForSingleTest(test.failure);
```

This would make intent clearer IMO. It would also remove the need for this weird `current` fallback object, since you call the function only if its parameter exists.